k4EDM4hep2LcioConv/src/k4EDM4hep2LcioConv.cpp

```diff
@@ -699,6 +699,20 @@ lcio::LCCollectionVec* convMCParticles(
   return mcparticles;
 }
 
+void convEventHeader(const edm4hep::EventHeaderCollection* const header_coll, lcio::LCEventImpl* const lcio_event)
+{
+  // the collection should be of length 1
+  if (header_coll->size() != 1) {
+    // TODO: somehow warn about this?
+    return;
+  }
```

Comment:
This could be made a pre-condition from the calling site (or at lest put it into the docstring for this function). I would not put out a warning (at least not to `stdout` or `stderr`), because that most likely interferes with Gaudis logging.